lib/config/rule-validator.js

```diff
@@ -187,6 +189,10 @@ class RuleValidator {
                     }`);
                 }
             }
+
+            if (mergedOptions.length) {
+                config.rules[ruleId] = [ruleOptions[0], ...mergedOptions];
```

Comment:
We shouldn't be changing anything in the config here. This is a validation function so it should just be inspecting values, not overwriting them.

I think it would be better to do this inside of `Config` for flat config. Specifically this function:
https://github.com/eslint/eslint/blob/151c965aec1c46000ac7dfc67a1c04802112aafc/lib/config/config.js#L129

For eslintrc, it's probably better to just do this before calling `ruleValidator.validate()`.